metrics/open-telemetry-backend/src/main/scala/sttp/client3/opentelemetry/OpenTelemetryBackend.scala

```diff
@@ -0,0 +1,98 @@
+package sttp.client3.opentelemetry
+
+import io.opentelemetry.api.OpenTelemetry
+import io.opentelemetry.api.common.{AttributeKey, Attributes}
+import io.opentelemetry.api.trace.{Span, SpanKind, StatusCode}
+import io.opentelemetry.context.Context
+import io.opentelemetry.context.propagation.{TextMapPropagator, TextMapSetter}
+import sttp.capabilities.Effect
+import sttp.client3._
+import sttp.monad.MonadError
+import sttp.monad.syntax._
+
+import scala.collection.mutable
+
+private class OpenTelemetryBackend[F[_], P](
+    delegate: SttpBackend[F, P],
+    openTelemetry: OpenTelemetry,
+    spanName: Option[String]
+) extends SttpBackend[F, P] {
+
+  private val tracer = openTelemetry.getTracer("sttp3-client", "1.0.0")
+  private implicit val _monad: MonadError[F] = responseMonad
+  type PE = P with Effect[F]
+
+  def send[T, R >: PE](request: Request[T, R]): F[Response[T]] = {
+    val carrier: mutable.Map[String, String] = mutable.Map().empty
+    val propagator: TextMapPropagator = openTelemetry.getPropagators.getTextMapPropagator
+    val setter: TextMapSetter[mutable.Map[String, String]] = (carrier, key, value) => carrier.update(key, value)
+    responseMonad
+      .eval {
+        val attributes = prepareBaseAttributes(request)
+        val span: Span = tracer
+          .spanBuilder(spanName.getOrElse(s"HTTP ${request.method.method}"))
+          .setSpanKind(SpanKind.CLIENT)
+          .setAllAttributes(attributes)
+          .startSpan
+        val scope = span.makeCurrent()
+        propagator.inject(Context.current(), carrier, setter)
+        (span, scope)
+      }
+      .flatMap { spanAndScope =>
+        val (span, scope) = spanAndScope
+        val label = requestLabel(request)
+        Span.current.addEvent(s"Start handling request: $label")
```

Comment:
hm reading through the opentelemtry docs, I don't think the events add any value above what's in the request attributes